mindmeld/text_preparation/tokenizers.py

```diff
@@ -0,0 +1,199 @@
+                    tokens.append(token)
+                token = {}
+                token_text = ""
+                continue
+            if (
+                token_text and cat[0] != unicodedata.category(token_text[-1])[0]
+            ) or cat == "Lo":
+                # if we hit a non-Latin character, close the token and restart buffer
+                if token and token_text:
+                    token["text"] = token_text
+                    tokens.append(token)
+                token = {"start": i}
+                token_text = char
+                continue
+            if not token_text:
+                token = {"start": i}
+            token_text += char
+
+        if token and token_text:
+            token["text"] = token_text
+            tokens.append(token)
+
+        return tokens
+
+
+class WhiteSpaceTokenizer(Tokenizer):
+    """A Tokenizer that splits text at spaces."""
+
+    def __init__(self):
+        """Initializes the WhtieSpaceTokenizer."""
+        pass
+
+    def tokenize(self, text):
+        """
+        Identify tokens in text and create normalized tokens that contain the text and start index.
+        Args:
+            text (str): The text to normalize
+        Returns:
+            tokens (List[str]): List of tokenized tokens.
+        """
+        tokens = []
+        token = {}
+        token_text = ""
+        for i, char in enumerate(text):
+            if char.isspace():
+                if token and token_text:
+                    token["text"] = token_text
+                    tokens.append(token)
+                token = {}
+                token_text = ""
+                continue
+            if not token_text:
+                token = {"start": i}
+            token_text += char
+
+        if token and token_text:
+            token["text"] = token_text
+            tokens.append(token)
+
+        return tokens
+
+
+class SpacyTokenizer(Tokenizer):
+    """A Tokenizer that splits text at spaces."""
+
+    def __init__(self, language, spacy_model_size):
+        """Initializes a SpacyTokenizer."""
+        self.spacy_model = SpacyModelFactory.get_spacy_language_model(
+            language, spacy_model_size
+        )
+
+    def tokenize(self, text):
+        """
+        Args:
+            text (str): Input text.
+        Returns:
+            tokens (List[str]): List of tokens created using a Spacy Language Model.
+        """
+        spacy_tokens = [token.text for token in self.spacy_model(text)]
+
+        start_index = 0
+        tokens = []
+        for token_text in spacy_tokens:
+            token = {
+                "start": start_index,
+                "text": token_text,
+            }
+            tokens.append(token)
+            start_index += len(token_text)
+        return tokens
+
+
+class TokenizerFactory:
+    """Tokenizer Factory Class"""
+
+    @staticmethod
+    def get_tokenizer(tokenizer, language=None, spacy_model_size="lg"):
```

Comment:
Why is "lg" the default? Did tokenization quality depend on model size?

Reply:
Good point, looks like Spacy's tokenization is rule based so I'll change this to "sm"